compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -281,7 +281,13 @@ import transform.SymUtils._
       val (foundStr, expectedStr) = Formatting.typeDiff(found2, expected2)(using printCtx)
       s"""|Found:    $foundStr
           |Required: $expectedStr""".stripMargin
-        + whereSuffix + postScript
+        + whereSuffix + postScript 
+
+    override def explain = 
+      val treeStr = inTree.map(x => s"\nTree: ${x.show}").getOrElse("")
+      treeStr + "\n" + err.whyNoMatchStr(found, expected)
```

Comment:
To avoid diverging from the parent's definition of explain, let's reuse it here:

```suggestion
      treeStr + "\n" + super.explain
```